src/chaos/controller.go

```diff
@@ -0,0 +1,210 @@
+package chaos
+
+import (
+	"bytes"
+	"context"
+	"fmt"
+	"log"
+	"os"
+	"strings"
+
+	"github.com/picodata/xk6-khorne/src/client"
+
+	v1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/apimachinery/pkg/apis/meta/v1/unstructured"
+	"k8s.io/apimachinery/pkg/runtime/schema"
+	"k8s.io/apimachinery/pkg/util/yaml"
+	"k8s.io/client-go/dynamic"
+)
+
+type ChaosController struct {
+	DynamicClient dynamic.Interface
+	Namespace     string
+}
+
+// NewController creates a new instance of ChaosController
+func NewController(namespace string) *ChaosController {
+	config, err := client.GetConfig()
+	if err != nil {
+		log.Fatal(err.Error())
+		return nil
+	}
+
+	dynamicClient, err := dynamic.NewForConfig(config)
+	if err != nil {
+		log.Fatalf("Failed to create Kubernetes client: %v", err)
+	}
+
+	return &ChaosController{dynamicClient, namespace}
+}
```

Comment:
nit: I don't think this will significantly affect performance, but I think it's still better to use a singleton here.

```suggestion
var chaosControllerSingltone *ChaosController

// NewController creates a new instance of ChaosController
func NewController(namespace string) *ChaosController {
	if chaosControllerSingltone != nil {
		return chaosControllerSingltone
	}

	config, err := client.GetConfig()
	if err != nil {
		log.Fatal(err.Error())
		return nil
	}

	dynamicClient, err := dynamic.NewForConfig(config)
	if err != nil {
		log.Fatalf("Failed to create Kubernetes client: %v", err)
	}

	chaosControllerSingltone = &ChaosController{dynamicClient, namespace}

	return chaosControllerSingltone
}
```